Improve the process and smell integration documentation. (#9)
HabchiSarra/Sniffer: README.md

    4. Based on the extracted commits order and the detected code smells, track the history of each code smell instance and store it in the `PostgreSQL` database.

## Integrate new smells

### Add a new code smell query

The easiest way to integrate a new code smell in this process would be to create a new definition in `SmellDetector`.
This means writing a `Neo4j` query relying on the metadata persisted by `SmellDetector`.
It may require to add some new metadata in the persisted model, but will not be much of a hassle to integrate.

This is most likely the easiest way, but the less sustainable if the `Paprika` model does not fit all your needs.

**TODO: Add links & details**

### Add smells in the SmellDetector database

If your smell detection are performed before persisting the model and already handled by a visitor pattern,
you may want to add this detection before persisting the model, since Spoon is already working with visitor.
It would then be possible to add new metadata in the `SmellDetector` database, to represent the detected code smells.

The point is that you will have less work to do to transform your code smell definition.-

> However, if you also want to keep our originally detected code smells you would still be required to write a new detection query for them.

Reviewer: I think my original statement was not clear enough, I was meaning something along the line:
```suggestion
However, if you want to track your new code smells you would be required to write a new detection query for them.
```
